sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/functions.scala

```diff
@@ -930,3 +930,327 @@ object HyperLogLogPlusPlus {
+    }
+  }
+}
+
+case class Variance(child: Expression, mutableAggBufferOffset: Int = 0,
+                    inputAggBufferOffset: Int = 0) extends SecondMoment(child) {
+
+  override def withNewMutableAggBufferOffset(newMutableAggBufferOffset: Int): ImperativeAggregate =
+    copy(mutableAggBufferOffset = newMutableAggBufferOffset)
+
+  override def withNewInputAggBufferOffset(newInputAggBufferOffset: Int): ImperativeAggregate =
+    copy(inputAggBufferOffset = newInputAggBufferOffset)
+
+  override def prettyName: String = "variance"
+
+  override protected val isBiased = false
+
+  override protected val isStd = false
+}
+
+case class VarianceSamp(child: Expression, mutableAggBufferOffset: Int = 0,
+                        inputAggBufferOffset: Int = 0) extends SecondMoment(child) {
+
+  override def withNewMutableAggBufferOffset(newMutableAggBufferOffset: Int): ImperativeAggregate =
+    copy(mutableAggBufferOffset = newMutableAggBufferOffset)
+
+  override def withNewInputAggBufferOffset(newInputAggBufferOffset: Int): ImperativeAggregate =
+    copy(inputAggBufferOffset = newInputAggBufferOffset)
+
+  override def prettyName: String = "variance_samp"
+
+  override protected val isBiased = false
+
+  override protected val isStd = false
+}
+
+case class VariancePop(child: Expression, mutableAggBufferOffset: Int = 0,
+                       inputAggBufferOffset: Int = 0) extends SecondMoment(child) {
+
+  override def withNewMutableAggBufferOffset(newMutableAggBufferOffset: Int): ImperativeAggregate =
+    copy(mutableAggBufferOffset = newMutableAggBufferOffset)
+
+  override def withNewInputAggBufferOffset(newInputAggBufferOffset: Int): ImperativeAggregate =
+    copy(inputAggBufferOffset = newInputAggBufferOffset)
+
+  override def prettyName: String = "variance_pop"
+
+  override protected val isBiased = true
+
+  override protected val isStd = false
+}
+
+case class Skewness(child: Expression, mutableAggBufferOffset: Int = 0,
+                    inputAggBufferOffset: Int = 0) extends CentralMomentAgg(child) {
+
+  override def withNewMutableAggBufferOffset(newMutableAggBufferOffset: Int): ImperativeAggregate =
+    copy(mutableAggBufferOffset = newMutableAggBufferOffset)
+
+  override def withNewInputAggBufferOffset(newInputAggBufferOffset: Int): ImperativeAggregate =
+    copy(inputAggBufferOffset = newInputAggBufferOffset)
+
+  override def prettyName: String = "skewness"
+
+  override protected val momentOrder = 3
+
+  override protected val sufficientMoments = Array(2, 3)
+
+  override def getStatistic(n: Double, moments: Array[Double]): Double = {
+    require(moments.length == sufficientMoments.length,
+      s"$prettyName requires two central moments, received: ${moments.length}")
+    val Array(m2, m3) = moments
+    if (n == 0.0 || m2 == 0.0) {
+      Double.NaN
+    } else {
+      math.sqrt(n) * m3 / math.sqrt(m2 * m2 * m2)
+    }
+  }
+}
+
+case class Kurtosis(child: Expression, mutableAggBufferOffset: Int = 0,
```

Comment:
* Need to document which Kurtosis we implemented.
* Chop down args to match Spark code style.